internal/stanza/frompdataconverter.go

```diff
@@ -0,0 +1,273 @@
+	wg sync.WaitGroup
+
+	logger *zap.Logger
+}
+
+func NewFromPdataConverter(workerCount int, logger *zap.Logger) *FromPdataConverter {
+	if logger == nil {
+		logger = zap.NewNop()
+	}
+	if workerCount <= 0 {
+		workerCount = int(math.Max(1, float64(runtime.NumCPU())))
+	}
+
+	return &FromPdataConverter{
+		workerChan:  make(chan fromConverterWorkerItem, workerCount),
+		entriesChan: make(chan []*entry.Entry),
+		stopChan:    make(chan struct{}),
+		logger:      logger,
+	}
+}
+
+func (c *FromPdataConverter) Start() {
+	c.logger.Debug("Starting log converter from pdata", zap.Int("worker_count", cap(c.workerChan)))
+
+	for i := 0; i < cap(c.workerChan); i++ {
+		c.wg.Add(1)
+		go c.workerLoop()
+	}
+}
+
+func (c *FromPdataConverter) Stop() {
+	c.stopOnce.Do(func() {
+		close(c.stopChan)
+		c.wg.Wait()
+		close(c.entriesChan)
+		close(c.workerChan)
+	})
+}
+
+// OutChannel returns the channel on which converted entries will be sent to.
+func (c *FromPdataConverter) OutChannel() <-chan []*entry.Entry {
+	return c.entriesChan
+}
+
+type fromConverterWorkerItem struct {
+	Resource       pdata.Resource
+	LogRecordSlice pdata.LogRecordSlice
+}
+
+// workerLoop is responsible for obtaining pdata logs from Batch() calls,
+// converting them to []*entry.Entry and sending them out
+func (c *FromPdataConverter) workerLoop() {
+	defer c.wg.Done()
+
+	for {
+		select {
+		case <-c.stopChan:
+			return
+
+		case workerItem, ok := <-c.workerChan:
+			if !ok {
+				return
+			}
+
+			select {
+			case c.entriesChan <- convertFromLogs(workerItem):
+			case <-c.stopChan:
+				return
+			}
+		}
+	}
+}
+
+// Batch takes in an set of pdata.Logs and sends it to an available worker for processing.
+func (c *FromPdataConverter) Batch(pLogs pdata.Logs) error {
+	for i := 0; i < pLogs.ResourceLogs().Len(); i++ {
+		rls := pLogs.ResourceLogs().At(i)
+		for j := 0; j < rls.ScopeLogs().Len(); j++ {
+			scope := rls.ScopeLogs().At(j)
+			item := fromConverterWorkerItem{
+				Resource:       rls.Resource(),
+				LogRecordSlice: scope.LogRecords(),
+			}
+			select {
+			case c.workerChan <- item:
+				continue
+			case <-c.stopChan:
+				return nil
+			}
+		}
+	}
+
+	return nil
+}
+
+// convertFromLogs converts the contents of a fromConverterWorkerItem into a slice of entry.Entry
+func convertFromLogs(workerItem fromConverterWorkerItem) []*entry.Entry {
+	result := make([]*entry.Entry, 0, workerItem.LogRecordSlice.Len())
+	for i := 0; i < workerItem.LogRecordSlice.Len(); i++ {
+		record := workerItem.LogRecordSlice.At(i)
+		entry := entry.Entry{}
+
+		entry.Resource = valueToMap(workerItem.Resource.Attributes())
+		convertFrom(record, &entry)
+		result = append(result, &entry)
+	}
+	return result
+}
+
+// ConvertFrom converts pdata.Logs into a slice of entry.Entry
+// To be used in a stateless setting like tests where ease of use is more
+// important than performance or throughput.
+func ConvertFrom(pLogs pdata.Logs) []*entry.Entry {
+	result := []*entry.Entry{}
```

Comment:
Even though not a performance optimized entry point, might as well:
```suggestion
	result := make([]*entry.Entry, 0, pLogs.LogRecordCount())
```